examples/interaction.rs

```diff
@@ -0,0 +1,292 @@
+                .spawn_bfs()
+                .report(&mut WriteReporter::new(&mut std::io::stdout()));
+
+            checker.assert_properties();
+        }
+
+        Some("explore") => {
+            let checker = model
+                .property(Expectation::Eventually, "success", |_, state| {
+                    state.actor_states.iter().any(|s| state_filter_success(s))
+                })
+                .checker()
+                .threads(num_cpus::get())
+                .target_max_depth(30); // To ensure termination of the checker, since the state space is very loosely bounded.
+
+            checker.serve("0:3000");
+        }
+
+        _ => {
+            println!("USAGE:");
+            println!("  ./interaction check");
+            println!("  ./interaction explore");
+        }
+    }
+    Ok(())
+}
+
+// We use this helper function to improve readability of check_discovery.
+// It also shows how to use the choice! macro for destructuring.
+fn state_filter_success(s: &Arc<choice![InputState, SupervisorState, CounterState]>) -> bool {
+    match s.as_ref() {
+        choice!(0 -> _v) => false,    // InputState
+        choice!(1 -> v) => v.success, // SupervisorState
+        choice!(2 -> _v) => false,    // CounterState
+        choice!(3 -> !) => false,     // Never
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub enum Msg {
+    SupervisorIncrementRequest(CounterSize),
+    SupervisorReportRequest(),
+    CounterReplyCount(CounterSize),
+}
+
+// We use a Generic BaseActor to pass State to on_start, but Actors could be built using independent structs.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct BaseActor<T> {
+    pub initial_state: T, //T needs to be a State Type
+}
+
+pub type CounterSize = u32;
+
+pub type Counter = BaseActor<CounterState>;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub struct CounterState {
+    pub addr: Id,
+    pub counter: CounterSize,
+}
+
+impl Actor for Counter {
+    type Msg = Msg;
+    type State = CounterState;
+    type Timer = InputTimer;
+
+    fn on_start(&self, _id: Id, _o: &mut Out<Self>) -> Self::State {
+        self.initial_state
+    }
+
+    fn on_msg(
+        &self,
+        _id: Id,
+        state: &mut Cow<Self::State>,
+        src: Id,
+        msg: Self::Msg,
+        o: &mut Out<Self>,
+    ) {
+        let (new_state, output_msgs) = match msg {
+            Self::Msg::SupervisorIncrementRequest(n) => (
+                CounterState {
+                    counter: state.counter + n,
+                    ..**state
+                },
+                Vec::new(),
+            ),
+
+            Self::Msg::SupervisorReportRequest() => (
+                **state,
+                vec![(src, Self::Msg::CounterReplyCount(state.counter))],
+            ),
+
+            _ => (**state, Vec::new()),
+        };
```

Comment:
I also think the example will be simpler if you just mutate the state and call `send` within each arm of the `match` statement for all the actors.